File: 0234-palindrome-linked-list/0234-palindrome-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    ListNode reverse(ListNode head)
    {
        ListNode cur=head,pre=null,next=null;
        while (cur!=null)
        {
            next=cur.next;
            cur.next=pre;
            pre=cur;
            cur=next;
        }
        return pre;
    }
    public boolean isPalindrome(ListNode head) {
        ListNode slow=head,fast=head;
        while(fast !=null && fast.next!=null)
        {
            slow=slow.next;
            fast=fast.next.next;
        }
        ListNode head2=reverse(slow);
        ListNode cur1=head,cur2=head2;
        while (cur1!=null && cur2!=null)
        {
            if (cur1.val!=cur2.val)
            {
                return false;
            }
            cur1=cur1.next;
            cur2=cur2.next;
        }
        return true;
        
    }
}

File: 0234-palindrome-linked-list/0234-palindrome-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        slow=head
        fast=head
        while fast.next and fast.next.next:
            slow=slow.next
            fast=fast.next.next
        cur=slow.next;
        prev=None
        while cur:
            next=cur.next
            cur.next=prev
            prev=cur
            cur=next
        f=head
        s=prev
        while s and f:
            if s.val !=f.val:
                return False
            f=f.next
            s=s.next
        return True
